Tests/CommandsTests/BuildCommandTests.swift

```diff
@@ -678,4 +679,36 @@ final class BuildCommandTests: CommandsTestCase {
             XCTAssertGreaterThan(codeCovFiles.count, 0)
         }
     }
+
+    // Test for GitHub Issue #6605
+    func testFatalErrorDisplayedOnlyOnceWhenSingleXCTestHasFatalErrorInBuildCompilaation() async throws {
+        log_info("GIVEN we have a Swift Package that has a fatalError building the tests")
```

Comment:
To me these kind of comments clutter up the the test logs as there isn't typically any output logged between these lines that they would be adding context to.

I'd like to remove these methods and just convert these lines to comments.

Reply:
I find them helpful on test failure as you have context as to the test intent. I'll remove them since they also appear on test cases that pass, and add to the clutter.

```
Test Case '-[CommandsTests.BuildCommandTests testFatalErrorDisplayedOnlyOnceWhenSingleXCTestHasFatalErrorInBuildCompilaation]' started.
Emitted by Test | /Users/bkhouri/Documents/git/public/swiftlang/swift-package-manager/Tests/CommandsTests/BuildCommandTests.swift:685 | INFO | GIVEN we have a Swift Package that has a fatalError building the tests
Emitted by Test | /Users/bkhouri/Documents/git/public/swiftlang/swift-package-manager/Tests/CommandsTests/BuildCommandTests.swift:687 | INFO | WHEN swift-build --build-tests is executed
Emitted by Test | /Users/bkhouri/Documents/git/public/swiftlang/swift-package-manager/Tests/CommandsTests/BuildCommandTests.swift:689 | INFO | THEN I expect a failure
Emitted by Test | /Users/bkhouri/Documents/git/public/swiftlang/swift-package-manager/Tests/CommandsTests/BuildCommandTests.swift:699 | DEBUG | stdout: "[0/1] Planning build\nBuilding for debugging...\n[0/7] /private/var/folders/1b/57vwczd16h733tf_gzsx7d8c0000gn/T/Miscellaneous_Errors_FatalErrorInSingleXCTest_TypeLibrary.tTCaUe/Miscellaneous_Errors_FatalErrorInSingleXCTest_TypeLibrary/.build/arm64-apple-macosx/debug/TypeLibraryPackageTests.derived/runner.swift\n[1/7] Write sources\n[4/7] Write swift-version-3DF09919E8911743.txt\n[6/9] Compiling TypeLibrary TypeLibrary.swift\n[7/9] Emitting module TypeLibrary\n[8/11] Emitting module TypeLibraryTests\n[9/11] Compiling TypeLibraryTests TypeLibraryTests.swift\n/private/var/folders/1b/57vwczd16h733tf_gzsx7d8c0000gn/T/Miscellaneous_Errors_FatalErrorInSingleXCTest_TypeLibrary.tTCaUe/Miscellaneous_Errors_FatalErrorInSingleXCTest_TypeLibrary/Tests/TypeLibraryTests/TypeLibraryTests.swift:10:15: error: division by zero\n 8 | func testExample() throws {\n 9 |     let x = 0\n10 |     let y = 1 / x\n   |               `- error: division by zero\n11 |     print(y)\n12 | }\n"
Emitted by Test | /Users/bkhouri/Documents/git/public/swiftlang/swift-package-manager/Tests/CommandsTests/BuildCommandTests.swift:700 | DEBUG | stderr: ""
Emitted by Test | /Users/bkhouri/Documents/git/public/swiftlang/swift-package-manager/Tests/CommandsTests/BuildCommandTests.swift:701 | DEBUG | number stdout matches: 0
Emitted by Test | /Users/bkhouri/Documents/git/public/swiftlang/swift-package-manager/Tests/CommandsTests/BuildCommandTests.swift:702 | DEBUG | number stderr matches: 0
Emitted by Test | /Users/bkhouri/Documents/git/public/swiftlang/swift-package-manager/Tests/CommandsTests/BuildCommandTests.swift:704 | INFO | AND a fatal error message is not printed to the console
/Users/bkhouri/Documents/git/public/swiftlang/swift-package-manager/Tests/CommandsTests/BuildCommandTests.swift:705: error: -[CommandsTests.BuildCommandTests testFatalErrorDisplayedOnlyOnceWhenSingleXCTestHasFatalErrorInBuildCompilaation] : XCTAssertEqual failed: ("0") is not equal to ("2") - We expected 1 math but found 0 occurrances
```

Reply:
I believe it could be interesting to print this kind of output on test failure. In a future PR it might be worth constructing a utility that takes a closure whose first argument is an `ObservabilitySystem` (already in SPM). Then you could write tests like

```swift
func testSomething() throws {
  withFailureLogging({ observer in 
    observer.emit(.log("starting test...))
    XCTFail("Oh no")
  })
}
```

Failures could be caught, the logs printed only then, and then the error rethrown.